compiler/rustc_query_system/src/query/job.rs

```diff
@@ -552,7 +552,9 @@ pub fn deadlock<D: DepKind>(query_map: QueryMap<D>, registry: &rayon_core::Regis
     // which in turn will wait on X causing a deadlock. We have a false dependency from
     // X to Y due to Rayon waiting and a true dependency from Y to X. The algorithm here
     // only considers the true dependency and won't detect a cycle.
-    assert!(found_cycle);
+    if !found_cycle {
+        panic!("deadlock detected");
+    }
```

Comment:
```suggestion
    assert!(found_cycle, "deadlock detected");
```

Reply:
I went with `panic!` for the cleaner error message here.

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=6546fb921d5da013fea2262e8ed17427
Assert seems to give the same message

Reply:
I probably confused it with the `assert_eq!` variants which does have the `assertion failed` prefix. I prefer the explicit `if` as the message doesn't directly describe the assertion condition anyway.